### DISCUSSION TITLE
Layout transition with framer motion

### DISCUSSION THREAD

**Discussion by martin-dimi** [xyflow/xyflow#2995] [Q&A]
Hey yall. 

I'm trying to transition from a layout which uses reactflow to display a tree graph to another "detail" page when a node is clicked. The idea is that upon clicking a node, the node (for me an image) expands and moves to fit the desired location in the new layout. For this I'm using `layoutId` from framer-motion.

However, I'm getting some undesirable effects. I have a suspicion this is due the node's positioning relative to the reactflow's viewport. 
![chrome-capture-2023-3-11 (1)](https://user-images.githubusercontent.com/25355177/231268286-c534fce9-98f4-41ce-9e80-82c568c80cfd.gif)

Would it be possible to get a node's coordinates and size (both in px) relative to the browser during an onClick event?

PS: removing `fitView` from `Editor` makes the issue more obvious. It's trying to adjust the position to the original location, afterwards fitView kicks in and readjust the new position.

**Comment by moklick** [comment]
The width and height values are stored in `node.width` and `node.height` (we are measuring the node and store that information there). The node coordinates relative to the your screen can be calculated by:

```js 
const x = node.position.x * transform[2] + transform[0];
const y = node.position.y * transform[2] + transform[1];
```
